crates/bevy_render/src/render_resource/storage_buffer.rs

```diff
@@ -0,0 +1,138 @@
+use std::{
+    num::NonZeroU64,
+    ops::{Deref, DerefMut},
+};
+
+use bevy_crevice::std430::{self, AsStd430, Std430};
+use bevy_utils::tracing::warn;
+use wgpu::{BindingResource, BufferBinding, BufferDescriptor, BufferUsages};
+
+use crate::renderer::{RenderDevice, RenderQueue};
+
+use super::Buffer;
+
+/// A helper for a storage buffer binding with a body, or a variable-sized array, or both.
+pub struct StorageBuffer<T: AsStd430, U: AsStd430> {
```

Comment:
I think this should be `StorageBuffer<T: AsStd430, U: AsStd430 = ()>`. That enables `StorageBuffer<SomeBody>` and `StorageBuffer<SomeBody, SomeListItem>`.
Or alternatively, we could consider flipping them so that `StorageBuffer<SomeListItem>` works. Given that people generally reach for storage buffers because they want unbounded lists, thats probably the better default.

Reply:
Yup, I can do that.

Reply:
Done.